homeassistant/components/media_player/__init__.py

```diff
@@ -174,51 +170,51 @@ def is_on(hass, entity_id=None):
     await component.async_setup(config)
 
     component.async_register_entity_service(
-        SERVICE_TURN_ON, MEDIA_PLAYER_SCHEMA,
+        SERVICE_TURN_ON, ENTITY_SERVICE_SCHEMA,
         'async_turn_on', [SUPPORT_TURN_ON]
     )
     component.async_register_entity_service(
-        SERVICE_TURN_OFF, MEDIA_PLAYER_SCHEMA,
+        SERVICE_TURN_OFF, ENTITY_SERVICE_SCHEMA,
         'async_turn_off', [SUPPORT_TURN_OFF]
     )
     component.async_register_entity_service(
-        SERVICE_TOGGLE, MEDIA_PLAYER_SCHEMA,
+        SERVICE_TOGGLE, ENTITY_SERVICE_SCHEMA,
         'async_toggle', [SUPPORT_TURN_OFF | SUPPORT_TURN_ON]
     )
     component.async_register_entity_service(
-        SERVICE_VOLUME_UP, MEDIA_PLAYER_SCHEMA,
+        SERVICE_VOLUME_UP, ENTITY_SERVICE_SCHEMA,
         'async_volume_up', [SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP]
     )
     component.async_register_entity_service(
-        SERVICE_VOLUME_DOWN, MEDIA_PLAYER_SCHEMA,
+        SERVICE_VOLUME_DOWN, ENTITY_SERVICE_SCHEMA,
         'async_volume_down', [SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP]
     )
     component.async_register_entity_service(
-        SERVICE_MEDIA_PLAY_PAUSE, MEDIA_PLAYER_SCHEMA,
+        SERVICE_MEDIA_PLAY_PAUSE, ENTITY_SERVICE_SCHEMA,
         'async_media_play_pause', [SUPPORT_PLAY | SUPPORT_PAUSE]
     )
     component.async_register_entity_service(
-        SERVICE_MEDIA_PLAY, MEDIA_PLAYER_SCHEMA,
+        SERVICE_MEDIA_PLAY, ENTITY_SERVICE_SCHEMA,
         'async_media_play', [SUPPORT_PLAY]
     )
     component.async_register_entity_service(
-        SERVICE_MEDIA_PAUSE, MEDIA_PLAYER_SCHEMA,
+        SERVICE_MEDIA_PAUSE, ENTITY_SERVICE_SCHEMA,
         'async_media_pause', [SUPPORT_PAUSE]
     )
     component.async_register_entity_service(
```

Comment:
We should allow passing a dictionary to `async_register_entity_service, and add this code to that function:

```python
if isinstance(schema, Dict):
    schema = ENTITY_SERVICE_SCHEMA.extend(schema)
```

Could clean up code a lot .

Reply:
Awesome. I'll take care of that in a separate PR.